psci/Main.hs

```diff
@@ -92,6 +129,267 @@ getCommand singleLineMode = handleInterrupt (return (Right Nothing)) $ do
     go :: [String] -> InputT m String
     go ls = maybe (return . unlines $ reverse ls) (go . (:ls)) =<< getInputLine "  "
 
+-- | Make a JavaScript bundle for the browser.
+bundle :: IO (Either Bundle.ErrorMessage String)
+bundle = runExceptT $ do
+  inputFiles <- liftIO (glob (".psci_modules" </> "node_modules" </> "*" </> "*.js"))
+  input <- for inputFiles $ \filename -> do
+    js <- liftIO (readUTF8File filename)
+    mid <- Bundle.guessModuleIdentifier filename
+    length js `seq` return (mid, js)
+  Bundle.bundle input [] Nothing "PSCI"
+
+-- TODO: use JMacro here?
```

Comment:
I haven't used JMacro, so I can't speak to its applicability here, but I have used [file-embed](http://hackage.haskell.org/package/file-embed) successfully for this sort of thing before.